Vagrantfile

```diff
@@ -5,13 +5,82 @@
 # configures the configuration version (we support older styles for
 # backwards compatibility). Please don't change it unless you know what
 # you're doing.
+
+
+def os_is_osx?
+  `uname`.strip == 'Darwin'
+end
+
+def default_if_osx
+  `route get default | grep interface`.split(" ").last
+end
+
+def default_if_linux
+  `/sbin/route | grep default`.split(" ").last
+end
+
+# Returns the name of the default interface
+def default_if
+  @default_if_memo ||= os_is_osx? ? default_if_osx : default_if_linux
+end
+
+def interface_is_bridge?(interface)
+  `/usr/sbin/brctl show | cut -f1 | grep '^#{interface}$'`.strip.length > 0
+end
+
+# While VBox on Linux allows bridging without a host device, there's no easy way
+# to check the provider from the Vagrantfile (provider-specific config blocks
+# still run on the wrong providers) so assume libvirt on linux deployments
+#
+# It's also worth noting that the Vagrant libvirt provider (KVM) typically uses
+# 'direct' type interfaces for public_network interfaces. While these *do* work
+# without a host bridge interface, the host VM is unable to route to IPs the VM
+# receives this way. See https://libvirt.org/formatnetwork.html#examplesDirect
+def host_bridge_available?
+  # Exploit the fact that puts returns nil, which is falsy
+  if os_is_osx?
+    puts "Bridged networking not implemented on OSX"
+  elsif ! default_if
+    puts "No default interface detected"
+  elsif ! File.file? "/usr/sbin/brctl"
+    puts "'brctl' tool not found. Have you installed bridge-utils"
+  elsif ! interface_is_bridge?(default_if)
+    warning_env = {
+      "COMMAND"    => "vagrant up --provider=libvirt",
+      "DEFAULT_IF" => default_if
+    }
+    puts system(warning_env, "bin/common/warn_no_bridge.sh")
+  else
+    true
+  end
+end
+
+def bridged_net_linux?()
+  @bridged_net_memo ||= ENV.fetch("VAGRANT_BRIDGED", false) && host_bridge_available?
+end
+
+# In theory, vbox could manage a bridged deployment if the interface is not wireless
+# But for now, we'll assume vbox is primarily used for osx deployments, which means
+# a wired interface may not be available
+def bridged_net_osx?
+  false
+end
+
+def bridged_net?
+  @bridged_net_memo ||= os_is_osx? ? bridged_net_linux? : bridged_net_osx?
+end
+
+def net_dhcp?
+  ENV.fetch("VAGRANT_DHCP", false) ? true : bridged_net?
```

Comment:
I'm using `ENV.include?` now for `VAGRANT_DHCP`: https://github.com/SUSE/scf/blob/harts-vagrant-use-public-network-rebased/Vagrantfile#L15. `VAGRANT_KVM_BRIDGE` and `VAGRANT_VBOX_BRIDGE` are now expected to have values (though I've still used `ENV.include?` to check if they're set)